packages/velo-external-db-core/lib/service/schema_aware_data.js

```diff
@@ -13,6 +14,7 @@ class SchemaAwareDataService {
     }
 
     async getById(collectionName, itemId) {
+        await this.validateFilter(collectionName, getByIdFilter(itemId))
```

Comment:
add test for this and also validate cases of invalid id, empty string/null value should throw an exception